dove/src/stdout.rs

```diff
@@ -0,0 +1,213 @@
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Yellow)))
+}
+
+/// print output with color settings
+fn print_with_color(text: &str, spec: &mut ColorSpec) -> Result<(), Error> {
+    let mut buffer = BufferWriter::stdout(get_stdout_color()).buffer();
+    buffer.set_color(spec)?;
+    write!(&mut buffer, "{}", text)?;
+    buffer.reset()?;
+    print(&String::from_utf8_lossy(buffer.as_slice()))
+}
+
+/// Prints output.
+#[macro_export]
+macro_rules! stdout {
+    ($($args:tt)*) => {
+        $crate::stdout::print(&format!($($args)*)).unwrap();
+    };
+}
+
+/// Prints output with a newline.
+#[macro_export]
+macro_rules! stdoutln {
+    () => ($crate::stdout::print!("\n").unwrap());
+    ($($args:tt)*) => {
+        $crate::stdout::println(&format!($($args)*)).unwrap();
+    }
+}
+
+/// Set the stream / buffer to write
+pub fn set_buffer<Out>(stdout: Out) -> Result<(), Error>
+where
+    Out: TBufWrite + Send,
+    Out: 'static,
+{
+    let flag = get_argv_flag_color(&stdout);
+
+    match STDOUT_COLOR.get() {
+        Some(mt) => {
+            let mut buff = mt.lock().unwrap();
+            *buff = Box::new(flag);
+            Ok(())
+        }
+        None => STDOUT_COLOR
+            .set(Mutex::new(Box::new(flag)))
+            .map_or_else(|_| Err(anyhow!("Failed to set STDOUT_COLOR")), |_| Ok(())),
+    }?;
+
+    match STDOUT_STREAM.get() {
+        Some(mt) => {
+            let mut buff = mt.lock().unwrap();
+            *buff = Box::new(stdout);
+            Ok(())
+        }
+        None => STDOUT_STREAM
+            .set(Mutex::new(Box::new(stdout)))
+            .map_or_else(|_| Err(anyhow!("Failed to set buffer output")), |_| Ok(())),
+    }
+}
+
+/// Get data from the buffer if a string was used
+pub fn get_buffer_value() -> Option<String> {
+    STDOUT_STREAM
+        .get()
+        .and_then(|mt| mt.lock().map_or(None, |bx| bx.value()))
+}
+
+/// get marker to highlight
+fn get_stdout_color() -> ColorChoice {
+    match STDOUT_COLOR.get() {
+        Some(mt) => *(mt.lock().unwrap()).clone(),
+        None => ColorChoice::Auto,
+    }
+}
+
+/// get flag --color=.. auto|always|never|ansi
+fn get_argv_flag_color<Out>(stdout: &Out) -> ColorChoice
+where
+    Out: TBufWrite + Send,
+    Out: 'static,
+{
+    let flag = std::env::args()
+        .find_map(|it| {
+            if it.to_lowercase().contains("--color=") {
+                Some((&it[8..]).to_lowercase())
+            } else {
+                None
+            }
+        })
+        .unwrap_or_else(|| "auto".to_string());
+
+    match flag.as_str() {
+        "always" => ColorChoice::Always,
+        "ansi" => ColorChoice::AlwaysAnsi,
+        "auto" => {
+            if stdout.is_stdout() && atty::is(atty::Stream::Stdout) {
+                ColorChoice::Auto
+            } else {
+                ColorChoice::Never
+            }
+        }
+        _ => ColorChoice::Never,
+    }
+}
+
+/// The stream / buffer to write
+pub trait TBufWrite {
```

Comment:
Please don't use `T` prefix for traits, I believe it is directly against [Rust naming convention](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html)